types/serverless-step-functions/index.d.ts

```diff
@@ -0,0 +1,310 @@
+    enabled: boolean;
+};
+
+type Definition = {
+    Comment?: string;
+    StartAt: string;
+    States: States;
+    ProcessorConfig?: {
+        Mode: 'INLINE' | 'DISTRIBUTED';
+    };
+};
+
+type States = {
+    [state: string]: Choice | Fail | Map | Task | Parallel | Pass | Wait | Succeed;
+};
+
+type StateBase = {
+    Catch?: Catcher[];
+    Retry?: Retrier[];
+    End?: boolean;
+    InputPath?: string;
+    Next?: string;
+    OutputPath?: string;
+    ResultPath?: string | null;
+    ResultSelector?: { [key: string]: string | { [key: string]: string } };
+    Type: string;
+    Comment?: string;
+};
+
+type ChoiceRuleComparison = {
+    Variable: string;
+    BooleanEquals?: boolean;
+    BooleanEqualsPath?: string;
+    IsBoolean?: boolean;
+    IsNull?: boolean;
+    IsNumeric?: boolean;
+    IsPresent?: boolean;
+    IsString?: boolean;
+    IsTimestamp?: boolean;
+    NumericEquals?: number;
+    NumericEqualsPath?: string;
+    NumericGreaterThan?: number;
+    NumericGreaterThanPath?: string;
+    NumericGreaterThanEquals?: number;
+    NumericGreaterThanEqualsPath?: string;
+    NumericLessThan?: number;
+    NumericLessThanPath?: string;
+    NumericLessThanEquals?: number;
+    NumericLessThanEqualsPath?: string;
+    StringEquals?: string;
+    StringEqualsPath?: string;
+    StringGreaterThan?: string;
+    StringGreaterThanPath?: string;
+    StringGreaterThanEquals?: string;
+    StringGreaterThanEqualsPath?: string;
+    StringLessThan?: string;
+    StringLessThanPath?: string;
+    StringLessThanEquals?: string;
+    StringLessThanEqualsPath?: string;
+    StringMatches?: string;
+    TimestampEquals?: string;
+    TimestampEqualsPath?: string;
+    TimestampGreaterThan?: string;
+    TimestampGreaterThanPath?: string;
+    TimestampGreaterThanEquals?: string;
+    TimestampGreaterThanEqualsPath?: string;
+    TimestampLessThan?: string;
+    TimestampLessThanPath?: string;
+    TimestampLessThanEquals?: string;
+    TimestampLessThanEqualsPath?: string;
+};
+
+type ChoiceRuleNot = {
+    Not: ChoiceRuleComparison;
+    Next: string;
+    Comment?: string;
+};
+
+type ChoiceRuleAnd = {
+    And: ChoiceRuleComparison[];
+    Next: string;
+    Comment?: string;
+};
+
+type ChoiceRuleOr = {
+    Or: ChoiceRuleComparison[];
+    Next: string;
+    Comment?: string;
+};
+
+type ChoiceRuleSimple = ChoiceRuleComparison & {
+    Next: string;
+    Comment?: string;
+};
+
+type ChoiceRule = ChoiceRuleSimple | ChoiceRuleNot | ChoiceRuleAnd | ChoiceRuleOr;
+
+interface Choice extends StateBase {
+    Type: 'Choice';
+    Choices: ChoiceRule[];
+    Default?: string;
+}
+
+interface Fail extends StateBase {
+    Type: 'Fail';
+    Cause?: string;
+    Error?: string;
+}
+
+interface Map extends StateBase {
+    Type: 'Map';
+    ItemsPath?: string;
+    Iterator: Definition;
+    ItemSelector?: {
+        [key: string]: string;
+    };
+    MaxConcurrency?: number;
+    Parameters?: {
+        [key: string]: string;
+    };
+}
```

Comment:
`Map` is missing the `ItemProcessor` field which was added in the Dec 2022 update.

There are some checks in [asl-validator](https://github.com/ChristopheBougere/asl-validator) to reject definitions that have both deprecated and new fields for the same state. i.e. a `Map` that defines both `Parameters` and `ItemSelector.` It would be good to encode this in the TS definition.